datafusion/functions-aggregate/Cargo.toml

```diff
@@ -44,6 +44,8 @@ arrow-schema = { workspace = true }
 datafusion-common = { workspace = true }
 datafusion-execution = { workspace = true }
 datafusion-expr = { workspace = true }
+datafusion-functions-aggregate-common = { workspace = true }
+datafusion-physical-expr = { workspace = true }
```

Comment:
I guess this is no longer an issue given the `datafusion-physical-expr` is somewhat *thin* now. We could move out `expressions` for minimum dependencies

Reply:
What do you mean by "thin"? I checked on this branch and here is what datafusion/physical-expr still has

(we could potentially move the groups accumulators into `datafusion-functions-aggegate-common` as well, as a follow on PR)

```
$ find datafusion/physical-expr/src
datafusion/physical-expr/src
datafusion/physical-expr/src/physical_expr.rs
datafusion/physical-expr/src/partitioning.rs
datafusion/physical-expr/src/analysis.rs
datafusion/physical-expr/src/aggregate
datafusion/physical-expr/src/aggregate/mod.rs
datafusion/physical-expr/src/aggregate/stats.rs
datafusion/physical-expr/src/aggregate/groups_accumulator
datafusion/physical-expr/src/aggregate/groups_accumulator/adapter.rs
datafusion/physical-expr/src/aggregate/groups_accumulator/mod.rs
datafusion/physical-expr/src/lib.rs
datafusion/physical-expr/src/functions.rs
datafusion/physical-expr/src/utils
datafusion/physical-expr/src/utils/guarantee.rs
datafusion/physical-expr/src/utils/mod.rs
datafusion/physical-expr/src/planner.rs
datafusion/physical-expr/src/intervals
datafusion/physical-expr/src/intervals/test_utils.rs
datafusion/physical-expr/src/intervals/mod.rs
datafusion/physical-expr/src/intervals/cp_solver.rs
datafusion/physical-expr/src/intervals/utils.rs
datafusion/physical-expr/src/equivalence
datafusion/physical-expr/src/equivalence/properties.rs
datafusion/physical-expr/src/equivalence/mod.rs
datafusion/physical-expr/src/equivalence/projection.rs
datafusion/physical-expr/src/equivalence/class.rs
datafusion/physical-expr/src/equivalence/ordering.rs
datafusion/physical-expr/src/window
datafusion/physical-expr/src/window/row_number.rs
datafusion/physical-expr/src/window/ntile.rs
datafusion/physical-expr/src/window/window_expr.rs
datafusion/physical-expr/src/window/lead_lag.rs
datafusion/physical-expr/src/window/cume_dist.rs
datafusion/physical-expr/src/window/nth_value.rs
datafusion/physical-expr/src/window/built_in.rs
datafusion/physical-expr/src/window/built_in_window_function_expr.rs
datafusion/physical-expr/src/window/aggregate.rs
datafusion/physical-expr/src/window/rank.rs
datafusion/physical-expr/src/window/mod.rs
datafusion/physical-expr/src/window/sliding_aggregate.rs
datafusion/physical-expr/src/expressions
datafusion/physical-expr/src/expressions/like.rs
datafusion/physical-expr/src/expressions/negative.rs
datafusion/physical-expr/src/expressions/literal.rs
datafusion/physical-expr/src/expressions/cast.rs
datafusion/physical-expr/src/expressions/binary.rs
datafusion/physical-expr/src/expressions/is_not_null.rs
datafusion/physical-expr/src/expressions/unknown_column.rs
datafusion/physical-expr/src/expressions/mod.rs
datafusion/physical-expr/src/expressions/is_null.rs
datafusion/physical-expr/src/expressions/try_cast.rs
datafusion/physical-expr/src/expressions/column.rs
datafusion/physical-expr/src/expressions/no_op.rs
datafusion/physical-expr/src/expressions/binary
datafusion/physical-expr/src/expressions/binary/kernels.rs
datafusion/physical-expr/src/expressions/case.rs
datafusion/physical-expr/src/expressions/in_list.rs
datafusion/physical-expr/src/expressions/not.rs
datafusion/physical-expr/src/scalar_function.rs
datafusion/physical-expr/src/math_expressions.rs
```

Reply:
I think half of them are `window` which will be moved to `functions-window-common` in the future. Another half are `equivalence`. However, I agree they are still unnecessary dependency for `functions-aggregate`. 😞  I think we could iteratively move `window` and maybe `equivalence` 🤔  out from `physical-expr` and keep most of expressions under `datafusion/physical-expr/src/expressions/` at the end if we expect minimum dependency for 3rd party.

Reply:
We can see if `window` and/or `equivalence` staying here causes any problems in practice. If so, I agree that they can be moved in the future. However, it is also possible that they do not pose any challenges in practice and can stay. We'll see soon enough 🚀

Reply:
> We can see if `window` and/or `equivalence` staying here causes any problems in practice. If so, I agree that they can be moved in the future. However, it is also possible that they do not pose any challenges in practice and can stay. We'll see soon enough 🚀

💯  agree

FWIW one project that would result in moving window functions would be https://github.com/apache/datafusion/issues/8709

I suspect that is a relatively smaller project than moving scalar or aggregates as there are so many fewer window functions